oak_functions/wasm/src/lib.rs

```diff
@@ -476,59 +565,13 @@ where
     }
 }
 
-/// A resolver function, mapping `oak_functions` host function names to an index and a type
-/// signature.
-fn oak_functions_resolve_func(field_name: &str) -> Option<(usize, wasmi::Signature)> {
-    // The types in the signatures correspond to the parameters from
-    // oak_functions_abi/src/lib.rs
-    let (index, expected_signature) = match field_name {
-        "read_request" => (
-            READ_REQUEST,
-            wasmi::Signature::new(
-                &[
-                    ABI_USIZE, // buf_ptr_ptr
-                    ABI_USIZE, // buf_len_ptr
-                ][..],
-                Some(ValueType::I32),
-            ),
-        ),
-        "write_response" => (
-            WRITE_RESPONSE,
-            wasmi::Signature::new(
-                &[
-                    ABI_USIZE, // buf_ptr
-                    ABI_USIZE, // buf_len
-                ][..],
-                Some(ValueType::I32),
-            ),
-        ),
-        "invoke" => (
-            INVOKE,
-            wasmi::Signature::new(
-                &[
-                    ABI_USIZE, // handle
-                    ABI_USIZE, // request_ptr
-                    ABI_USIZE, // request_len
-                    ABI_USIZE, // response_ptr_ptr
-                    ABI_USIZE, // response_len_ptr
-                ][..],
-                Some(ValueType::I32),
-            ),
-        ),
-        _ => return None,
-    };
-
-    Some((index, expected_signature))
-}
-
 /// A helper function to move between our specific result type `Result<(), OakStatus>` and the
-/// `wasmi` specific result type `Result<Option<wasmi::RuntimeValue>, wasmi::Trap>`, mapping:
-/// - `Ok(())` to `Ok(Some(OakStatus::Ok))`
-/// - `Err(x)` to `Ok(Some(x))`
-fn from_oak_status_result(
-    result: Result<(), OakStatus>,
-) -> Result<Option<wasmi::RuntimeValue>, wasmi::Trap> {
-    let oak_status_from_result = result.map_or_else(|x: OakStatus| x, |()| OakStatus::Ok);
-    let wasmi_value = wasmi::RuntimeValue::I32(oak_status_from_result as i32);
-    Ok(Some(wasmi_value))
+/// `wasmi` specific result type `Result<i32, wasmi::Trap>`.
+// TODO(mschett): Changed result type from Option<i32> to i32. Check implications.
+fn from_oak_status_result(result: Result<(), OakStatus>) -> Result<i32, wasmi::core::Trap> {
+    let oak_status: OakStatus = match result {
+        Ok(()) => OakStatus::Ok,
+        Err(oak_status) => oak_status,
+    };
```

Comment:
Could this be `let oak_status = result.err().unwrap_or(OakStatus::Ok)` or something like that?